source/graphImplementation/graph.py

```diff
@@ -1,53 +1,141 @@
+        if self.__mother is None and parent.__gender == "female":
+            parent.add_child(self)
+
+    def add_spouse(self, spouse):
+        """
+        adds spouse to person, adds new parent to children
+        adds children to spouse
+        """
+        if self.__spouse is None:
+            self.__spouse = spouse
+            self.__spouse.__spouse = self
+            if spouse.__gender == "male":
+                for i in self.__children:
+                    i.__father = spouse
+            else:
+                for i in self.__children:
+                    i.__mother = spouse
+            self.__spouse.__children = self.__children
+
+    def get_person(self, name):
+        """
+        looking for a person by name and returns Node with him
+        """
+        for i in self.__children:
+            if i.get_name == name:
+                return i
+            if i.get_person(name).get_name == name:
+                return i.get_person(name)
+        return None
 
 
 class Graph(object):
 
-    def __init__(self, graph_dict = None):
-        """ initializes a graph object
-            If no dictionary or None is given,
-            an empty dictionary will be used
-        """
-        if graph_dict == None:
-            graph_dict = {}
-        self.__graph_dict = graph_dict
-
-    def vertices(self):
-        """ returns the vertices of a graph """
-        return list(self.__graph_dict.keys())
-
-    def edges(self):
-        """ returns the edges of a graph """
-        return self.__generate_edges()
-
-    def add_vertex(self, vertex):
-        """ If the vertex "vertex" is not in
-            self.__graph_dict, a key "vertex" with an empty
-            list as a value is added to the dictionary.
-            Otherwise nothing has to be done.
-        """
-        if vertex not in self.__graph_dict:
-            self.__graph_dict[vertex] = []
-
-    def add_edge(self, edge):
-        """ assumes that edge is of type set, tuple or list;
-            between two vertices can be multiple edges!
-        """
-        edge = set(edge)
-        (vertex1, vertex2) = tuple(edge)
-        if vertex1 in self.__graph_dict:
-            self.__graph_dict[vertex1].append(vertex2)
-        else:
-            self.__graph_dict[vertex1] = [vertex2]
-
-    def __generate_edges(self):
-        """ A static method generating the edges of the
-            graph "graph". Edges are represented as sets
-            with one (a loop back to the vertex) or two
-            vertices
-        """
-        edges = []
-        for vertex in self.__graph_dict:
-            for neighbour in self.__graph_dict[vertex]:
-                if {neighbour, vertex} not in edges:
-                    edges.append({vertex, neighbour})
-        return edges
\ No newline at end of file
+    def __init__(self, root):
+        self.__root = root
+
+    def get_root(self):
+        return self.__root
+
+    def get_person(self, name):
```

Comment:
@margarita0303 , it's written with an assumption that `__root` is a parent of everyone in the tree. It can not find spouses or sisters/brothers of the person `__root`.